### DISCUSSION TITLE
can material object receive properties from non-material objects

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#24432] [Q&A General]
Hi All,

In Materials System, Material objects produce properties and other objects (including materials) consume these properties. Is it possible to input variable values calculated in the other objects such Interfacekernels or BCs objects into a Material object?
Please help.  S. Thomas

**Comment by GiudGiud** [comment]
Hello

typically not. BCs and interface kernels are consumers of properties and variables, not producers

you should look to move the computation of the quantities you care about to materials. 
Which quantities are you trying to compute in a BC / InterfaceKernel?

Guillaume
